WordPress/Classes/Services/ReaderCardService.swift

```diff
@@ -3,44 +3,65 @@ import Foundation
 class ReaderCardService {
     private let service: ReaderPostServiceRemote
     private let coreDataStack: CoreDataStack
+    private lazy var syncContext: NSManagedObjectContext = {
+        return coreDataStack.newDerivedContext()
+    }()
 
     init(service: ReaderPostServiceRemote = ReaderPostServiceRemote(wordPressComRestApi: WordPressComMockrestApi()),
          coreDataStack: CoreDataStack = ContextManager.shared) {
         self.service = service
         self.coreDataStack = coreDataStack
     }
 
-    func fetch(success: @escaping (Int, Bool) -> Void, failure: @escaping (Error?) -> Void) {
+    func fetch(page: Int = 1, success: @escaping (Int, Bool) -> Void, failure: @escaping (Error?) -> Void) {
         service.fetchCards(for: ["foo", "bar"],
-                           success: { [unowned self] cards in
-                            let syncContext = self.coreDataStack.newDerivedContext()
+                           success: { [weak self] cards in
 
-                            syncContext.perform {
-                                cards.forEach { remoteCard in
-                                    guard remoteCard.type != .unknown else {
-                                        return
-                                    }
+                            guard let syncContext = self?.syncContext else {
+                                return
+                            }
+
+                            self?.syncContext.perform {
```

Comment:
Shouldn't this be:

```suggestion
                            syncContext.perform {
```

?